buckets-mdapi/src/gc/delete.rs

```diff
@@ -78,13 +78,43 @@ pub(crate) fn action(
 }
 
 fn do_delete(
-    _payload: &DeleteGarbagePayload,
+    payload: &DeleteGarbagePayload,
     conn: &mut PostgresConnection,
     metrics: &RegisteredMetrics,
     log: &Logger,
 ) -> Result<(), String> {
     let mut txn = (*conn).transaction().map_err(|e| e.to_string())?;
 
+    // Read the current batch id
+    let batch_id = sql::txn_query(
+        sql::Method::GarbageBatchIdGet,
+        &mut txn,
+        gc::get_garbage_batch_id_sql(),
+        &[],
+        metrics,
+        log,
+    )
+    .map_err(|e| e.to_string())
+    .and_then(|batch_id_rows| {
+        gc::handle_batch_id_result(batch_id_rows.as_ref())
+    })?;
+
+    // The batch id serves as a means to avoid uninentionally deleting a batch
+    // of garbage that may not have bee fully cleaned up yet. This could happen
```

Comment:
Typo in comment: "been fully cleaned up yet."

Reply:
Done